src/improvements/tasks/MultisigTask.sol

```diff
@@ -1050,9 +1050,64 @@ abstract contract MultisigTask is Test, Script {
 
 abstract contract L2TaskBase is MultisigTask {
     using EnumerableSet for EnumerableSet.AddressSet;
+    using stdToml for string;
+
+    struct StateOverride {
+        address target;
+        KeyValueOverride keyValueOverride;
```

Comment:
Would it be difficult to make this an array, similar to how it's handled in [base-contracts](https://github.com/ethereum-optimism/base-org-contracts/blob/7cca28be0f94134f9de1c6f409779b57d5f269b7/script/universal/Simulation.sol#L16)? 

```suggestion
        KeyValueOverride[] keyValueOverrides;
```

If it's not easy, then I think we can skip it for upgrade 13, but we'll very likely need it eventually.

Reply:
Actually, I just noticed that we're still using base's `Simulation.sol` in this file, and there's even a [reference](https://github.com/ethereum-optimism/superchain-ops/blob/646559e5c260a03bc22fd5d034c17d6cf142d1ae/src/improvements/tasks/MultisigTask.sol#L663C59-L663C83) to the `Simulation.StateOverride` type in `printTenderlySimulationLink()`. 

Can we: 
- just use those types
- parse the toml into those types
- modify `printTenderlySimulationLink()` to accept the overrides as an argument or read them from a state variable where they get pushed to?

Reply:
I made it an array: https://github.com/ethereum-optimism/superchain-ops/pull/727/commits/007e2702f8b187925c3b54aed091c391f1c57985 

Right now, in the `config.toml` file the UX is a little janky because users have to supply well-formed (padded) bytes32 hex strings. When users supply them correctly it works great. When they don't they're not parsed correctly and it leads to errors.

Reply:
I've added this functionality to the `printTenderlySimulationLink` function. I noticed that we had some default state overrides being applied (threshold, nonce, owner mapping) already. I had to make sure these still get applied but if the user added the same slots to their config, the users config would take preference. I then combin the arrays. Let me know what you think.

Reply:
> I noticed that we had some default state overrides being applied (threshold, nonce, owner mapping) already. 

Is that something we can easily add a test for?

Reply:
I've added a bunch of tests for this. The default state overrides where being created and then inserted into the tenderly URL, they aren't being applied to the simulation (unless the user specifies them).

Reply:
I've refactored this PR quite a bit since this discussion.